internal/authentication/ldap_user_provider.go

```diff
@@ -265,15 +305,22 @@ func (p *LDAPUserProvider) getUserProfile(conn LDAPConnection, inputUsername str
 	return &userProfile, nil
 }
 
-func (p *LDAPUserProvider) resolveGroupsFilter(inputUsername string, profile *ldapUserProfile) (string, error) { //nolint:unparam
-	inputUsername = p.ldapEscape(inputUsername)
+func (p *LDAPUserProvider) resolveGroupsFilter(inputUsername string, profile *ldapUserProfile) (groupFilter string, err error) { //nolint:unparam
+	groupFilter = p.configuration.GroupsFilter
 
-	// The {input} placeholder is replaced by the users username input.
-	groupFilter := strings.ReplaceAll(p.configuration.GroupsFilter, "{input}", inputUsername)
+	if p.groupsFilterReplacementInput {
+		// The {input} placeholder is replaced by the users username input.
+		groupFilter = strings.ReplaceAll(p.configuration.GroupsFilter, "{input}", p.ldapEscape(inputUsername))
+	}
 
 	if profile != nil {
-		groupFilter = strings.ReplaceAll(groupFilter, "{username}", ldap.EscapeFilter(profile.Username))
-		groupFilter = strings.ReplaceAll(groupFilter, "{dn}", ldap.EscapeFilter(profile.DN))
+		if p.groupsFilterReplacementUsername {
+			groupFilter = strings.ReplaceAll(groupFilter, "{username}", ldap.EscapeFilter(profile.Username))
+		}
+
+		if p.groupsFilterReplacementDN {
+			groupFilter = strings.ReplaceAll(groupFilter, "{dn}", ldap.EscapeFilter(profile.DN))
+		}
```

Comment:
gobench indicates this adds at most 2ns to this function, but potentially saves 100-150ns per check for instances where the replacement doesn't exist.